hyperdrive-cli/commands/stakewise/utils/utils.go

```diff
@@ -18,45 +16,41 @@ func printUploadError(err error) {
 
 // Upload deposit data to the server
 func UploadDepositData(sw *client.StakewiseClient) (bool, error) {
+	// Initial attempt to upload all deposit data
 	fmt.Println("Uploading deposit data to the NodeSet server...")
-	response, err := sw.Api.Nodeset.UploadDepositData(false)
+	response, err := sw.Api.Nodeset.UploadDepositData()
 	if err != nil {
 		printUploadError(err)
 		return false, nil
 	}
 
 	data := response.Data
 	newKeyCount := len(data.UnregisteredPubkeys)
-	if newKeyCount == 0 {
+
+	if newKeyCount == 0 && data.SufficientBalance {
```

Comment:
Move this into the check on line 46 (see other comments first).

Reply:
resolved